Makefile

```diff
@@ -31,6 +32,12 @@ services: args?=up -d
 services: python
 	@tox -qe docker-compose -- $(args)
 
+.PHONY: db
+db: args?=upgrade head
+db: python
+	@tox -qqe dev --run-command 'initdb conf/development.ini'
+	@tox -qe dev --run-command 'alembic -c conf/alembic.ini $(args)'
```

Comment:
The first command has `-qq` so that tox doesn't print out "commands succeeded congratulations :)" after the first command, because then it would print it out twice. The second command has just one `-q`, the same as every other `tox` command in our `Makefile`'s, so tox _does_ print out its summary of the second command. If the first command fails tox will fail without running the second command and will print out the error from the first command

Reply:
The mechanism we are using to run the script is the same as `devdata`, but it's a bit obscure. I think we could make it clearer they are the same by having the lines look the same. So either change the `devdata` line to use `--run command` or to remove it here.

```
tox -qqe dev --run-command 'initdb conf/development.ini'

# vs.

tox -qe dev -- devdata conf/development.ini
tox -qe dev -- pshell conf/development.ini
```

Reply:
I think `--run-command` is the correct usage.

The `posargs` form (`tox -e <ENV> -- <POSARGS>`) is for when you want to provide an easy way to just replace part of one command that the env would run, and the env has to have explicit support for it in `tox.ini`. For example the tests command in `tox.ini` is `tests: coverage run -m pytest -v -Werror {posargs:tests/unit/}`, which means that to run just one test you can do `tox tests/unit/foo/bar.py` (no `-e tests` is needed because it's the default). Any _other_ commands in the tests env (aside from the command containing the `{posargs}` will still be run unmodified as well.

`--run-command`, on the other hand, completely replaces _all) of the env's commands from `tox.ini`, and is more suitble for cases like devdata, alembic and pshell, where we're reusing the `dev` env but to run a different command. Any `pre_commands` or `post_commands` for the env in `tox.ini` will still be run though, only `commands` are replaced.

The reason why various things in `Makefile`'s still use posargs when they should probably be using `--run-command` is that they were written before we had added the run-command plugin

Reply:
Added a commit to tidy this up